src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/SuccessfulRetryTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ServiceControlCompatibilityTests
+{
+    using Autofac;
+    using NServiceBus;
+    using NUnit.Framework;
+    using System;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class SuccessfulRetryTests : SqlScTest
+    {
+        [TestCaseSource(nameof(AllTransports))]
+        public async Task Can_successfully_retry_a_failed_message(Type transportDetailType)
+        {
+            var transportDetails = StartUp(transportDetailType);
+            await RunTest(transportDetails);
+        }
+
+        async Task RunTest(ITransportDetails transportDetails)
+        {
+            var handlerInstance = new TestMessageHandler();
+            var builder = new ContainerBuilder();
+            builder.RegisterInstance(handlerInstance);
+            var container = builder.Build();
```

Comment:
the arrange could be just the creation of the handler + endpoint, registration of the container, etc. is just noise and is not directly related to the test logic (why should you have to change the test codes if you use another container).

RE the state, would it be possible to send the flag via the Context to the handler?